src/Adapter/Product/AdminProductDataUpdater.php

```diff
@@ -139,7 +140,11 @@ public function duplicateProductIdList(array $productIdList)
             throw new UpdateProductException('Cannot duplicate many requested products', 5004);
         }
 
-        return true;
+        if (count($productIdList) > 1 && count($duplicateProducts) < 1) {
```

Comment:
I don't understand this condition.
So, if you have only one `$productIdList` and `$duplicateProducts` is empty,  it can be false?
If you try to find invalid duplicate. there is already a condition just above :thinking: 

Regarding the interface, you're returning the list of duplicated products, it breaks the interface which is waiting for a boolean :(